src/clib/pioc_support.c

```diff
@@ -93,19 +94,57 @@ int PIOc_strerror(int pioerr, char *errmsg)
 int PIOc_set_log_level(int level)
 {
 #if PIO_ENABLE_LOGGING
-    char log_filename[NC_MAX_NAME];
-
     /* Set the log level. */
     pio_log_level = level;
+#endif /* PIO_ENABLE_LOGGING */
+    return PIO_NOERR;
+}
 
-    /* Create a filename with the rank in it. */
-    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
-    sprintf(log_filename, "pio_log_%d.txt", my_rank);
+/** Initialize logging
+ * Open log file, if not opened yet, or
+ * increment ref count if already open
+ */
+void pio_init_logging(void )
```

Comment:
In the future, if you wish to have less changes in review, you should file a github issue before making changes, so that we agree as to what is supposed to be happening.

I agree there are some bugs with the log files, but you have introduced a lot of unnecessary code to deal with them.